src/plugins/gmx/gmxplugin.cpp

```diff
@@ -0,0 +1,185 @@
+/*
+ * GMX Tiled Plugin
+ * Copyright 2016, Jones Blunt <mrjonesblunt@gmail.com>
+ *
+ * This file is part of Tiled.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the Free
+ * Software Foundation; either version 2 of the License, or (at your option)
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "gmxplugin.h"
+
+#include "map.h"
+#include "tile.h"
+#include "tilelayer.h"
+#include "mapobject.h"
+#include "objectgroup.h"
+
+#include <QDir>
+#include <QFile>
+#include <QFileInfo>
+#include <QXmlStreamWriter>
+
+using namespace Tiled;
+using namespace Gmx;
+
+GmxPlugin::GmxPlugin()
+{
+}
+
+bool GmxPlugin::write(const Map *map, const QString &fileName)
+{
+    QFile file(fileName);
+    if (!file.open(QIODevice::WriteOnly | QIODevice::Text)) {
+        mError = tr("Could not open file for writing.");
+        return false;
+    }
+
+    QXmlStreamWriter stream(&file);
+
+    stream.setAutoFormatting(true);
+    stream.setAutoFormattingIndent(2);
+
+    stream.writeComment("This Document is generated by GameMaker, if you edit it by hand then you do so at your own risk!");
+
+    stream.writeStartElement("room");
+    stream.writeStartElement("caption");
+    stream.writeEndElement();
+
+    stream.writeTextElement("width", QString::number(map->tileWidth() * map->width()));
+
+    stream.writeTextElement("height", QString::number(map->tileHeight() * map->height()));
+
+    stream.writeTextElement("vsnap", QString::number(map->tileHeight()));
+    stream.writeTextElement("hsnap", QString::number(map->tileWidth()));
+
+    stream.writeTextElement("isometric", QString::number((map->orientation() == Map::Orientation::Isometric ? 1 : 0)));
+
+    stream.writeStartElement("instances");
+
+    uint objectId = 0u;
+    foreach (const Layer *layer, map->layers()) {
+
+        if (layer->layerType() != Layer::ObjectGroupType) continue;
+        const ObjectGroup *objectLayer = static_cast<const ObjectGroup*>(layer);
+
+        foreach(const MapObject *object, objectLayer->objects()) {
+            stream.writeStartElement("instance");
+
+            stream.writeAttribute("objName", object->name());
+            stream.writeAttribute("x", QString::number((int)(object->x())));
+            stream.writeAttribute("y", QString::number((int)(object->y() - object->height())));
+
+            stream.writeAttribute("id", QString::number(++objectId));
```

Comment:
Is there a reason you can't use `object->id()` here? Does it need to be a consecutive numbering?